jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -16,105 +16,78 @@
 
 package io.prometheus.jmx;
 
-import io.prometheus.jmx.common.http.ConfigurationException;
 import io.prometheus.jmx.common.http.HTTPServerFactory;
-import io.prometheus.metrics.exporter.httpserver.HTTPServer;
+import io.prometheus.jmx.common.opentelemetry.OpenTelemetryExporterFactory;
 import io.prometheus.metrics.instrumentation.jvm.JvmMetrics;
 import io.prometheus.metrics.model.registry.PrometheusRegistry;
 import java.io.File;
 import java.lang.instrument.Instrumentation;
 import java.net.InetAddress;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 
 public class JavaAgent {
 
-    public static final String CONFIGURATION_REGEX =
-            "^(?:((?:[\\w.-]+)|(?:\\[.+])):)?"
-                    + // host name, or ipv4, or ipv6 address in brackets
-                    "(\\d{1,5}):"
-                    + // port
-                    "(.+)"; // config file
-
-    private static final String DEFAULT_HOST = "0.0.0.0";
-
-    private static HTTPServer httpServer;
-
-    public static void agentmain(String agentArgument, Instrumentation instrumentation)
-            throws Exception {
+    /**
+     * Java agent main
+     *
+     * @param agentArgument agentArgument
+     * @param instrumentation instrumentation
+     */
+    public static void agentmain(String agentArgument, Instrumentation instrumentation) {
         premain(agentArgument, instrumentation);
     }
 
-    public static void premain(String agentArgument, Instrumentation instrumentation)
-            throws Exception {
+    /**
+     * Java agent premain
+     *
+     * @param agentArgument agentArgument
+     * @param instrumentation instrumentation
+     */
+    public static void premain(String agentArgument, Instrumentation instrumentation) {
         try {
-            Config config = parseConfig(agentArgument);
+            Arguments arguments = Arguments.parse(agentArgument);
+            File file = new File(arguments.getFilename());
 
             new BuildInfoMetrics().register(PrometheusRegistry.defaultRegistry);
             JvmMetrics.builder().register(PrometheusRegistry.defaultRegistry);
-            new JmxCollector(new File(config.file), JmxCollector.Mode.AGENT)
+            new JmxCollector(new File(arguments.getFilename()), JmxCollector.Mode.AGENT)
                     .register(PrometheusRegistry.defaultRegistry);
 
-            String host = config.host != null ? config.host : DEFAULT_HOST;
-
-            httpServer =
-                    new HTTPServerFactory()
-                            .createHTTPServer(
-                                    InetAddress.getByName(host),
-                                    config.port,
-                                    PrometheusRegistry.defaultRegistry,
-                                    new File(config.file));
+            switch (arguments.getMode()) {
+                case HTTP:
+                    {
```

Comment:
The curly brackets `{` `}` are not needed here.

Reply:
The braces are the result of using the `spotless-maven-plugin` and configured Google Java Format for code formatting.

Reference: https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used

Reply:
Since there are only two modes, I refactored the code to use an `if` / `else`.

The use of the `spotless-maven-plugin` along with Google Java Format requires braces even with a single statement.

https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used